File: drug_service/src/main/java/ru/bmstu/cp/rsoi/drug/model/GenericResponse.java
package ru.bmstu.cp.rsoi.drug.model;

import org.springframework.context.support.DefaultMessageSourceResolvable;
import org.springframework.validation.ObjectError;

import java.util.List;
import java.util.stream.Collectors;

public class GenericResponse {

    private String message;
    private String type;

    public GenericResponse(final String message) {
        super();
        this.message = message;
    }

    public GenericResponse(String message, String type) {
        this.message = message;
        this.type = type;
    }

    public GenericResponse(List<ObjectError> allErrors, String error) {
        this.type = error;
        this.message = allErrors
                .stream()
                .map(DefaultMessageSourceResolvable::getDefaultMessage)
                .collect(Collectors.joining(System.lineSeparator()));
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }
}

File: session_service/src/main/kotlin/ru/bmstu/cp/rsoi/session/model/GenericResponse.kt
package ru.bmstu.cp.rsoi.session.model

import org.springframework.validation.ObjectError
import java.util.stream.Collectors

class GenericResponse {
    var error_description: String
    var error: String? = null

    constructor(message: String) : super() {
        error_description = message
    }

    constructor(message: String, type: String?) {
        error_description = message
        error = type
    }

    constructor(allErrors: List<ObjectError>, error: String?) {
        this.error = error
        error_description = allErrors
            .stream()
            .map { obj: ObjectError -> obj.defaultMessage }
            .collect(Collectors.joining(System.lineSeparator()))
    }

}